builtin/aws/ec2/platform.go

```diff
@@ -243,7 +246,36 @@ func (p *Platform) Deploy(
 
 	var instances []*string
 
+	// activityTicker periodically checks the ASG activity API to look for failure
+	// or cancellation
+	activityTicker := time.NewTicker(10 * time.Second)
+	defer activityTicker.Stop()
+
 	for {
+		select {
+		case <-activityTicker.C:
+			dsa, err := as.DescribeScalingActivities(&autoscaling.DescribeScalingActivitiesInput{
+				AutoScalingGroupName: aws.String(serviceName),
+			})
+			if err != nil {
+				return nil, err
+			}
+
+			// check scaling activities, capture any non-nil status code and messages
+			// to check for failures
+			var lastStatusCode, lastStatusMessage string
+			if len(dsa.Activities) > 0 {
+				lastStatusCode = aws.StringValue(dsa.Activities[0].StatusCode)
+				lastStatusMessage = aws.StringValue(dsa.Activities[0].StatusMessage)
+			}
+
+			if lastStatusCode == autoscaling.ScalingActivityStatusCodeFailed ||
+				lastStatusCode == autoscaling.ScalingActivityStatusCodeCancelled {
+				return nil, status.Errorf(codes.FailedPrecondition, "error setting up autoscaling group. Last status code (%s): \n Message: %s", lastStatusCode, lastStatusMessage)
+			}
```

Comment:
Would it make sense to put an else branch here and log the `lastStatusCode` at a DEBUG/TRACE level? Or does the AWS client already give us that in its logs 🤔

Reply:
We discussed internally and decided against the mentioned `else` clause here. Resolving this, thanks